### PR TITLE
Add compatibility with new schema

### LINKED ISSUE #3: Audit report is not conform with new security-report schema
We are using a Gitlab Enterprise Edition v16.8.1-ee instance and viewing the audit report is not possible in Gitlab.
Looks like the report is not anymore conform with the JSON schema -> https://gitlab.com/gitlab-org/security-products/security-report-schemas/-/blob/master/dist/dependency-scanning-report-format.json

So the UI shows:
grafik

Any chance to update the report version and add the missing required key?

### PR DESCRIPTION
**Pull request by roblabla**
Adds compatibility with schema format 15.0.7

Fixes #3

### CODE REVIEW

**Comment by dcoles** [comment]
Hi @roblabla,

Thank you for starting on this PR! I haven't checked our GitLab EE instance, but I suspect it might be broken on our pipeline too.

I'd be happy to take a look when you feel this PR is ready, or let me know if you have any questions.

**Comment by roblabla** [comment]
Tested it and integrated it at $WORK, where we're currently running 16.8.3-ee, and it works great.

This is ready for review.

File: src/main.rs [hunk 1]
```diff
@@ -30,18 +38,47 @@ fn main() -> anyhow::Result<()> {
...
 
+
     let report = report::Report {
         version: REPORT_VERSION.to_string(),
         vulnerabilities: report_vulnerabilities(&vulnerabilities),
+        scan: report::Scan {
+            analyzer: report::Analyzer {
+                id: SCANNER_ID.to_string(),
+                name: SCANNER_NAME.to_string(),
+                url: None,
```
Comment (dcoles) [inline src/main.rs:62]: From my reading of the GitLab Security Scanner terminology, I think the [Analyzer](https://docs.gitlab.com/ee/user/application_security/terminology/#analyzer) would be `gitlab-cargo-audit` while the [Scanner](https://docs.gitlab.com/ee/user/application_security/terminology/#scanner) would be `cargo-audit` would be the actual scanner. Does that sound right to you?

File: src/main.rs [hunk 2]
```diff
@@ -30,18 +38,47 @@ fn main() -> anyhow::Result<()> {
...
+                vendor: report::Vendor {
+                    name: SCANNER_VENDOR.to_string(),
+                },
+                version: env!("CARGO_PKG_VERSION").to_string(),
+            },
+            scanner: report::Scanner {
+                id: SCANNER_ID.to_string(),
+                name: SCANNER_NAME.to_string(),
+                url: None,
```
Comment (dcoles) [inline src/main.rs:71]: Would you mind adding `"https://crates.io/crates/cargo-audit"` as the `url` of the `scanner`.

File: src/report.rs [hunk 3]
```diff
@@ -8,69 +8,141 @@ use serde::Serialize;
...
+pub enum ScanStatus {
+    Success,
+    Failure,
+}
+
+impl Default for ScanStatus {
+    fn default() -> Self {
+        ScanStatus::Success
+    }
+}
```
Comment (dcoles) [inline src/report.rs:42]: You can now [derive Default on enums](https://doc.rust-lang.org/std/default/trait.Default.html#enums) and set the default using the `#[default]` attribute. I see you have a few places you have introduced new enums with a `Default` implementation.

**Review by dcoles — APPROVED** [review]
This looks great. Can you address the comments then I'll be happy to merge it.

**Comment by dcoles** [comment]
Actually, since they're easy to address I'll merge it as it currently is! Thanks for the PR!

**Pull request merged**